dask_cuda/tests/test_dask_cuda_worker.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import print_function, division, absolute_import
+
+import pytest
+pytest.importorskip('requests')
+
+import sys, os
+from time import sleep
+from toolz import first
+
+from distributed import Client
+from distributed.metrics import time
+from distributed.utils import sync, tmpfile
+from distributed.utils_test import (popen, slow, terminate_process,
+                                    wait_for_port)
+from distributed.utils_test import loop  # noqa: F401
+
+
+def test_nanny_worker_ports(loop):
+    os.environ["CUDA_VISIBLE_DEVICES"] = "2,3,7,8"
```

Comment:
The way to make this clean is to use either a try-finally block or perhaps the following context manager

```python
from contextlib import contextmanager

@contextmanager
def env(**kwargs):
    old = {k: os.environ.get(k, '--missing--') for k in kwargs}
    for k, v in kwargs.items():
        os.environ[k] = v

    yield # give control to the rest of the block

    for k, v in old.items():
        if v == '--missing--':
            del os.environ[k]
        else:
            os.environ[k] = v
```

Then you would use this as follows:

```python
with env(CUDA_VISIBLE_DEVICES='2,3,7,8'):
    with popen(...)
        ...
```